model/service.go

```diff
@@ -44,15 +45,28 @@ type Endpoint struct {
 	// Address of the endpoint, typically an IP address
 	Address string `json:"ip_address,omitempty"`
 	// Port on the host address
-	Port int `json:"port,omitempty"`
+	Port int `json:"port"`
+	// Name of the port classifies ports for a single service
+	Name string `json:"name,omitempty"`
+	// Protocol for the port: TCP, UDP (default is TCP)
+	Protocol Protocol `json:"protocol,omitempty"`
 }
 
+// Protocol defines network protocols for ports
+type Protocol string
+
+const (
+	ProtocolHTTP Protocol = "HTTP"
```

Comment:
Yes, but I also want to have duplicate HTTP ports (which can happen in k8s), so I check for a prefix and keep port name around.